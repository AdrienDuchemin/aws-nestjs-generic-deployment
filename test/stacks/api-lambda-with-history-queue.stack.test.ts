import { expect as expectCDK, SynthUtils, countResources, haveResource } from '@aws-cdk/assert'
import { Code } from '@aws-cdk/aws-lambda'
import { App } from '@aws-cdk/core'
import { ApiLambdaWithHistoryQueueStack } from '../../src/stacks/api-lambda-with-history-queue.stack'

describe('ApiLambdaWithHistoryStack', () => {
  let stack: ApiLambdaWithHistoryQueueStack

  beforeAll(()=> {
    stack = new ApiLambdaWithHistoryQueueStack(new App(), 'ApiLambdaWithHistoryStack', {
      lambdaProps: {
        code: Code.fromInline('lambda'),
        handler: 'handler',
      },
      historyQueueProps: {
        fifo: true
      }
    })
  })

  it('should create two Lambda', () => {
    expectCDK(stack).to(countResources("AWS::Lambda::Function", 2))
  })

  it('should create one Api Gateway', () => {
    expectCDK(stack).to(countResources("AWS::ApiGateway::RestApi", 1))
  })

  it('should create one Queue', () => {
    expectCDK(stack).to(countResources("AWS::SQS::Queue", 1))
    expectCDK(stack).to(haveResource("AWS::SQS::Queue",{
      FifoQueue: true,
    }))
  })
  
  it('should create one Table', () => {
    expectCDK(stack).to(countResources("AWS::DynamoDB::Table", 1))
  })

  it('should not have changed', () => {
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot()
  })
})