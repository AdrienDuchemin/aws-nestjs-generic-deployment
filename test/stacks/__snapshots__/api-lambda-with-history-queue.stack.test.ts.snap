// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiLambdaWithHistoryStack should not have changed 1`] = `
Object {
  "Outputs": Object {
    "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApiEndpoint30678BB8": Object {
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            "https://",
            Object {
              "Ref": "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApi3348E911",
            },
            ".execute-api.",
            Object {
              "Ref": "AWS::Region",
            },
            ".",
            Object {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            Object {
              "Ref": "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApiDeploymentStageprod35E3E8ED",
            },
            "/",
          ],
        ],
      },
    },
  },
  "Parameters": Object {
    "AssetParameters6e299eb822cce449ea66aea74a07269cec00aa22054e7820f61d319cb0031e05ArtifactHash6241674D": Object {
      "Description": "Artifact hash for asset \\"6e299eb822cce449ea66aea74a07269cec00aa22054e7820f61d319cb0031e05\\"",
      "Type": "String",
    },
    "AssetParameters6e299eb822cce449ea66aea74a07269cec00aa22054e7820f61d319cb0031e05S3Bucket3765D8AA": Object {
      "Description": "S3 bucket for asset \\"6e299eb822cce449ea66aea74a07269cec00aa22054e7820f61d319cb0031e05\\"",
      "Type": "String",
    },
    "AssetParameters6e299eb822cce449ea66aea74a07269cec00aa22054e7820f61d319cb0031e05S3VersionKey59FE3835": Object {
      "Description": "S3 key for asset version \\"6e299eb822cce449ea66aea74a07269cec00aa22054e7820f61d319cb0031e05\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "ApiLambdaWithHistoryStackHistoryQueuefifoApiLambdaWithHistoryStackApiLambdaWithHistoryStackHistoryLambda8E3C3EC2": Object {
      "DependsOn": Array [
        "ApiLambdaWithHistoryStackHistoryQueuefifoApiLambdaWithHistoryStackApiLambdaWithHistoryStackHistoryLambdaServiceRoleDefaultPolicy9D3CDB44",
        "ApiLambdaWithHistoryStackHistoryQueuefifoApiLambdaWithHistoryStackApiLambdaWithHistoryStackHistoryLambdaServiceRole6AE1221A",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters6e299eb822cce449ea66aea74a07269cec00aa22054e7820f61d319cb0031e05S3Bucket3765D8AA",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters6e299eb822cce449ea66aea74a07269cec00aa22054e7820f61d319cb0031e05S3VersionKey59FE3835",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters6e299eb822cce449ea66aea74a07269cec00aa22054e7820f61d319cb0031e05S3VersionKey59FE3835",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "TABLE_NAME": Object {
              "Ref": "ApiLambdaWithHistoryStackHistoryQueuefifoApiLambdaWithHistoryStackApiLambdaWithHistoryStackHistoryTable86C1BAB5",
            },
          },
        },
        "FunctionName": "ApiLambdaWithHistoryStackHistoryLambda",
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaWithHistoryStackHistoryQueuefifoApiLambdaWithHistoryStackApiLambdaWithHistoryStackHistoryLambdaServiceRole6AE1221A",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
        "Timeout": 60,
      },
      "Type": "AWS::Lambda::Function",
    },
    "ApiLambdaWithHistoryStackHistoryQueuefifoApiLambdaWithHistoryStackApiLambdaWithHistoryStackHistoryLambdaServiceRole6AE1221A": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "ApiLambdaWithHistoryStackHistoryQueuefifoApiLambdaWithHistoryStackApiLambdaWithHistoryStackHistoryLambdaServiceRoleDefaultPolicy9D3CDB44": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "sqs:ReceiveMessage",
                "sqs:ChangeMessageVisibility",
                "sqs:GetQueueUrl",
                "sqs:DeleteMessage",
                "sqs:GetQueueAttributes",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "ApiLambdaWithHistoryStackHistoryQueuefifoC33E3D0B",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": Array [
                "dynamodb:BatchGetItem",
                "dynamodb:GetRecords",
                "dynamodb:GetShardIterator",
                "dynamodb:Query",
                "dynamodb:GetItem",
                "dynamodb:Scan",
                "dynamodb:BatchWriteItem",
                "dynamodb:PutItem",
                "dynamodb:UpdateItem",
                "dynamodb:DeleteItem",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "ApiLambdaWithHistoryStackHistoryQueuefifoApiLambdaWithHistoryStackApiLambdaWithHistoryStackHistoryTable86C1BAB5",
                    "Arn",
                  ],
                },
                Object {
                  "Ref": "AWS::NoValue",
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "aWithHistoryStackHistoryQueuefifoApiLambdaWithHistoryStackApiLambdaWithHistoryStackHistoryLambdaServiceRoleDefaultPolicy9D3CDB44",
        "Roles": Array [
          Object {
            "Ref": "ApiLambdaWithHistoryStackHistoryQueuefifoApiLambdaWithHistoryStackApiLambdaWithHistoryStackHistoryLambdaServiceRole6AE1221A",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "ApiLambdaWithHistoryStackHistoryQueuefifoApiLambdaWithHistoryStackApiLambdaWithHistoryStackHistoryLambdaSqsEventSourceApiLambdaWithHistoryStackApiLambdaWithHistoryStackHistoryQueuefifo9D7A44B7E410628C": Object {
      "Properties": Object {
        "EventSourceArn": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaWithHistoryStackHistoryQueuefifoC33E3D0B",
            "Arn",
          ],
        },
        "FunctionName": Object {
          "Ref": "ApiLambdaWithHistoryStackHistoryQueuefifoApiLambdaWithHistoryStackApiLambdaWithHistoryStackHistoryLambda8E3C3EC2",
        },
      },
      "Type": "AWS::Lambda::EventSourceMapping",
    },
    "ApiLambdaWithHistoryStackHistoryQueuefifoApiLambdaWithHistoryStackApiLambdaWithHistoryStackHistoryTable86C1BAB5": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "AttributeDefinitions": Array [
          Object {
            "AttributeName": "id",
            "AttributeType": "S",
          },
        ],
        "KeySchema": Array [
          Object {
            "AttributeName": "id",
            "KeyType": "HASH",
          },
        ],
        "ProvisionedThroughput": Object {
          "ReadCapacityUnits": 5,
          "WriteCapacityUnits": 5,
        },
        "TableName": "ApiLambdaWithHistoryStackHistoryTable",
      },
      "Type": "AWS::DynamoDB::Table",
      "UpdateReplacePolicy": "Retain",
    },
    "ApiLambdaWithHistoryStackHistoryQueuefifoC33E3D0B": Object {
      "Properties": Object {
        "FifoQueue": true,
        "QueueName": "ApiLambdaWithHistoryStackHistoryQueue.fifo",
      },
      "Type": "AWS::SQS::Queue",
    },
    "ApiLambdaWithHistoryStackLambda2CB65481": Object {
      "DependsOn": Array [
        "ApiLambdaWithHistoryStackLambdaServiceRoleDefaultPolicyF04B8A12",
        "ApiLambdaWithHistoryStackLambdaServiceRole20D45CC2",
      ],
      "Properties": Object {
        "Code": Object {
          "ZipFile": "lambda",
        },
        "Environment": Object {
          "Variables": Object {
            "QUEUE_URL": Object {
              "Ref": "ApiLambdaWithHistoryStackHistoryQueuefifoC33E3D0B",
            },
          },
        },
        "FunctionName": "ApiLambdaWithHistoryStackLambda",
        "Handler": "handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaWithHistoryStackLambdaServiceRole20D45CC2",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
        "Timeout": 30,
      },
      "Type": "AWS::Lambda::Function",
    },
    "ApiLambdaWithHistoryStackLambdaAllowEventRuleApiLambdaWithHistoryStackApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRuleDC0C7F25A5E3062C": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaWithHistoryStackLambda2CB65481",
            "Arn",
          ],
        },
        "Principal": "events.amazonaws.com",
        "SourceArn": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRuleAAAE9F02",
            "Arn",
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApi3348E911": Object {
      "Properties": Object {
        "Name": "ApiLambdaWithHistoryStackRestApi",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApiANY5CF59881": Object {
      "Properties": Object {
        "AuthorizationType": "NONE",
        "HttpMethod": "ANY",
        "Integration": Object {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                Object {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                Object {
                  "Fn::GetAtt": Array [
                    "ApiLambdaWithHistoryStackLambda2CB65481",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApi3348E911",
            "RootResourceId",
          ],
        },
        "RestApiId": Object {
          "Ref": "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApi3348E911",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApiANYApiPermissionApiLambdaWithHistoryStackApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApiA20D9C77ANY090BB308": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaWithHistoryStackLambda2CB65481",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApi3348E911",
              },
              "/",
              Object {
                "Ref": "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApiDeploymentStageprod35E3E8ED",
              },
              "/*/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApiANYApiPermissionTestApiLambdaWithHistoryStackApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApiA20D9C77ANYAADDC376": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaWithHistoryStackLambda2CB65481",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApi3348E911",
              },
              "/test-invoke-stage/*/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApiAccount0BBE79D3": Object {
      "DependsOn": Array [
        "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApi3348E911",
      ],
      "Properties": Object {
        "CloudWatchRoleArn": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApiCloudWatchRoleE6F2A7B6",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
    },
    "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApiCloudWatchRoleE6F2A7B6": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApiDeploymentFB1A0E0383a2f8e5d6da5a2e94099d4832e453f2": Object {
      "DependsOn": Array [
        "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApiproxyANY84846DD1",
        "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApiproxyAC2794C8",
        "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApiANY5CF59881",
      ],
      "Properties": Object {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": Object {
          "Ref": "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApi3348E911",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApiDeploymentStageprod35E3E8ED": Object {
      "Properties": Object {
        "DeploymentId": Object {
          "Ref": "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApiDeploymentFB1A0E0383a2f8e5d6da5a2e94099d4832e453f2",
        },
        "RestApiId": Object {
          "Ref": "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApi3348E911",
        },
        "StageName": "prod",
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApiproxyAC2794C8": Object {
      "Properties": Object {
        "ParentId": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApi3348E911",
            "RootResourceId",
          ],
        },
        "PathPart": "{proxy+}",
        "RestApiId": Object {
          "Ref": "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApi3348E911",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApiproxyANY84846DD1": Object {
      "Properties": Object {
        "AuthorizationType": "NONE",
        "HttpMethod": "ANY",
        "Integration": Object {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                Object {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                Object {
                  "Fn::GetAtt": Array [
                    "ApiLambdaWithHistoryStackLambda2CB65481",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": Object {
          "Ref": "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApiproxyAC2794C8",
        },
        "RestApiId": Object {
          "Ref": "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApi3348E911",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApiproxyANYApiPermissionApiLambdaWithHistoryStackApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApiA20D9C77ANYproxy4156CEBA": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaWithHistoryStackLambda2CB65481",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApi3348E911",
              },
              "/",
              Object {
                "Ref": "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApiDeploymentStageprod35E3E8ED",
              },
              "/*/{proxy+}",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApiproxyANYApiPermissionTestApiLambdaWithHistoryStackApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApiA20D9C77ANYproxy8618A2AB": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaWithHistoryStackLambda2CB65481",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRestApi3348E911",
              },
              "/test-invoke-stage/*/{proxy+}",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "ApiLambdaWithHistoryStackLambdaApiLambdaWithHistoryStackRuleAAAE9F02": Object {
      "Properties": Object {
        "ScheduleExpression": "rate(5 minutes)",
        "State": "ENABLED",
        "Targets": Array [
          Object {
            "Arn": Object {
              "Fn::GetAtt": Array [
                "ApiLambdaWithHistoryStackLambda2CB65481",
                "Arn",
              ],
            },
            "Id": "Target0",
            "Input": "{\\"ping\\":true}",
          },
        ],
      },
      "Type": "AWS::Events::Rule",
    },
    "ApiLambdaWithHistoryStackLambdaServiceRole20D45CC2": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "ApiLambdaWithHistoryStackLambdaServiceRoleDefaultPolicyF04B8A12": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "sqs:SendMessage",
                "sqs:GetQueueAttributes",
                "sqs:GetQueueUrl",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "ApiLambdaWithHistoryStackHistoryQueuefifoC33E3D0B",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "ApiLambdaWithHistoryStackLambdaServiceRoleDefaultPolicyF04B8A12",
        "Roles": Array [
          Object {
            "Ref": "ApiLambdaWithHistoryStackLambdaServiceRole20D45CC2",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
}
`;
